scene/resources/surface_tool.cpp

```diff
@@ -729,6 +729,20 @@ Ref<ArrayMesh> SurfaceTool::commit(const Ref<ArrayMesh> &p_existing, uint64_t p_
 
 	Array a = commit_to_arrays();
 
+	// Disable compression if all z euqals 0 (the mesh is 2D)
+	Vector<Vector3> mesh_vertices = a[RS::ARRAY_VERTEX];
+	bool is_mesh_2d = true;
+
+	for (int i = 0; i < mesh_vertices.size(); i++) {
+		if (!Math::is_zero_approx(mesh_vertices[i].z)) {
+			is_mesh_2d = false;
```

Comment:
```suggestion
			is_mesh_2d = false;
			break;
```
Early cancel out of loop.

Reply:
Thanks!